### DISCUSSION TITLE
Why use Immediately Invoked Function Expression to create router?

### DISCUSSION THREAD

**Discussion by Toshinaki** [edwinhern/express-typescript-2024#192] [Q&A] [answered]
Hi, @edwinhern 
Just found this great repo, and trying to modify my existing project based on it.

I'm not that familiar with backend typescript. I noticed that you're using IIFE to create routers
https://github.com/edwinhern/express-typescript-2024/blob/master/src/api/user/userRouter.ts#L14

Why not instead just this
```
const userRouter = express.Router();
...
export const userRouter;
```

**Comment by edwinhern** [comment]
I wasn't thinking properly lol. Working on refactor for this

**Reply by edwinhern** [reply, marked as answer]
- Updated router, service, and repo files

[#231](https://github.com/edwinhern/express-typescript-2024/pull/231))
